monai/eval_savers/nifti_saver.py

```diff
@@ -0,0 +1,103 @@
+
+        """
+        self.output_dir = output_dir
+        self.output_postfix = output_postfix
+        self.output_ext = output_ext
+
+    @staticmethod
+    def _create_file_basename(postfix, input_file_name, folder_path, data_root_dir=""):
+        """
+        Utility function to create the path to the output file based on the input
+        filename (extension is added by lib level writer before writing the file)
+
+        Args:
+            postfix (str): output name's postfix
+            input_file_name (str): path to the input image file
+            folder_path (str): path for the output file
+            data_root_dir (str): if not empty, it specifies the beginning parts of the input file's
+          absolute path. This is used to compute `input_file_rel_path`, the relative path to the file from
+          `data_root_dir` to preserve folder structure when saving in case there are files in different
+          folders with the same file names.
+        """
+
+        # get the filename and directory
+        filedir, filename = os.path.split(input_file_name)
+
+        # jettison the extension to have just filename
+        filename, ext = os.path.splitext(filename)
+        while ext != "":
+            filename, ext = os.path.splitext(filename)
+
+        # use data_root_dir to find relative path to file
+        filedir_rel_path = ""
+        if data_root_dir:
+            filedir_rel_path = os.path.relpath(filedir, data_root_dir)
+
+        # sub-folder path will be original name without the extension
+        subfolder_path = os.path.join(folder_path, filedir_rel_path, filename)
+        if not os.path.exists(subfolder_path):
+            os.makedirs(subfolder_path)
+
+        # add the sub-folder plus the postfix name to become the file basename in the output path
+        return os.path.join(subfolder_path, filename + "_" + postfix)
+
+    def save(self, data, meta_data):
+        """Save data into Nifti format file. The metadata should have the following keys:
+            - ``'filename_or_obj'`` -- for output file name creation, corresponding to filename or object
+            - ``'original_affine'`` (optional) for data orientation handling
+            - ``'affine'`` (optional) for data output affine.
+
+        args:
+            data (Tensor or ndarray): target data content that save into Nifti format file.
+            meta_data (dict): the meta data information corresponding to the data.
+        """
+        filename = meta_data['filename_or_obj']
+        original_affine = meta_data.get('original_affine', None)
+        affine = meta_data.get('affine', None)
+
+        if torch.is_tensor(data):
+            data = data.detach().cpu().numpy()
+        filename = self._create_file_basename(self.output_postfix, filename, self.output_dir)
+        filename = '{}{}'.format(filename, self.output_ext)
+        # change data to "channel last" format and write to nifti format file
+        data = np.moveaxis(data, 0, -1)
```

Comment:
need to clarify that we assume a channel first format for `data` in the docstring

Reply:
Added docstring in the latest patch.
Thanks.